compiler/rustc_lint/src/noop_method_call.rs

```diff
@@ -114,17 +114,19 @@ impl<'tcx> LateLintPass<'tcx> for NoopMethodCall {
 
         let orig_ty = expr_ty.peel_refs();
 
-        if receiver_ty == expr_ty {
+        if receiver_ty == expr_ty
+            && matches!(
+                name,
+                sym::noop_method_borrow | sym::noop_method_clone | sym::noop_method_deref
+            )
+        {
```

Comment:
We already match on the trait above this, so why would this be needed?

Reply:
Adding `Clone::clone` as a diagnostic item is getting picked up here somehow.

Reply:
I don't understand the issue. Please reproduce this using a test. Or add a commit which shows why this happens (not necessarily needs to be a part of this PR)

Reply:
The linked PR's build failure is because of this.

Reply:
The assumption that a diagnostic item being found means it's one of the diagnostic items the lint cares about is wrong.

Reply:
Are you sure that this is the underlying cause of the build failure of that PR? As far as I can see, that PR already contains these changes and still has errors. Are you sure that those errors are false positives?

Reply:
The error's are at stage1, which is before the new compiler with those changes is built.

Reply:
A test has been added that would also fail, but isn't the same cause. I assume the actual issue has to do with the clone impl on tuples being a builtin.